mpi-proxy-split/lower-half/munmap.c

```diff
@@ -108,13 +93,23 @@ removeMappings(void *addr, size_t len)
 int
 __wrap___munmap (void *addr, size_t len)
 {
-  int rc = __real___munmap(addr, len);
+  int rc = mprotect(addr, len, PROT_NONE);
   if (rc == 0) {
     removeMappings(addr, len);
   }
   return rc;
 }
 
+int
+munmap(void *addr, size_t len)
+{
+  if (addr < lh_info.memRange.start || addr >= lh_info.memRange.end) {
```

Comment:
Please add:
```
    char *msg[] = "*** Lower half called munmap, and it's not from the reserved memRange.";
    write(2, msg, sizeof(msg));
```
Originally, we were sloppy, in order to work quickly.  But now, we want to document any unusual behavior so that we can easily review the logic.